go/chat/attachment_crypt.go

```diff
@@ -1,45 +1,112 @@
 package chat
 
-import "io"
+import (
+	"crypto/rand"
+	"errors"
+	"io"
+
+	"github.com/agl/ed25519"
+	"github.com/keybase/client/go/chat/signencrypt"
+	"github.com/keybase/client/go/libkb"
+)
 
 type Encrypter interface {
 	// EncryptedLen returns the number of bytes that the ciphertext of
 	// size plaintext bytes will be.
 	EncryptedLen(size int) int
 
 	// Encrypt takes a plaintext reader and returns a ciphertext reader.
-	Encrypt(plaintext io.Reader) (ciphertext io.Reader)
+	// It generates new keys every time it is called.
+	Encrypt(plaintext io.Reader) (ciphertext io.Reader, err error)
+
+	// EncryptKey returns the ephemeral key that was used during Encrypt.
```

Comment:
Maybe clarify that `EncryptKey` and `VerifyKey` return the _last_ key that was used by `Encrypt`?

I guess it could be an option to return those two keys as additional values from `Encrypt`, rather than having the separate method calls, to get rid of the statefulness here? Totally your preference though, no more safety issues that I can see.